docs/src/references/c8y-configuration-management.md

```diff
@@ -163,3 +279,232 @@ Note that:
+       
+### The child device uploads its configuration file list on start and on change
+
+The configuration files for a child device `$CHILD_DEVICE_ID` are defined by
+the file `$TEDGE_CONFIG_DIR/c8y/$CHILD_DEVICE_ID/c8y-configuration-plugin.toml`.
+The content of this file is provided by the child-device configuration software
+but must be known by the `c8y_configuration_plugin`
+that needs to store an up-to-date version of each `$CHILD_DEVICE_ID/c8y-configuration-plugin.toml`.
+
+Acting as a proxy between the cloud and the child devices,
+the `c8y_configuration_plugin` can simply take a copy of these files
+when transferred between the cloud and the child devices.
+* For these copies on the fly nothing is expected for the child-device configuration software.
+* A first special case is added to have the `c8y_configuration_plugin` requesting itself
+  a snapshot of the configuration for a child device if missing
+  (i.e. there is no `c8y-configuration-plugin.toml` for that device)
+* A second special case is also required to let the child device be able to send update
+  of its `$CHILD_DEVICE_ID/c8y-configuration-plugin.toml` on creation and updates.
+
+1. On reception of a `successful` status message
+   related to the `"type": "c8y-configuration-plugin"`,
+   received either for download on `tedge/$CHILD_DEVICE_ID/commands/req/config_update`
+   or for upload on `tedge/$CHILD_DEVICE_ID/commands/req/config_snapshot`.
+   1. The `c8y_configuration_plugin` takes a copy of the transferred file
+      that is under `$TEDGE_HTTP_ROOT/$CHILD_DEVICE_ID/config_update/c8y-configuration-plugin`
+      or `$TEDGE_HTTP_ROOT/$CHILD_DEVICE_ID/config_snapshop/c8y-configuration-plugin`
+      (depending on the file has been pushed by the cloud or the child device).
+   1. The `c8y_configuration_plugin` puts this copy under
+      `$TEDGE_CONFIG_DIR/c8y/$CHILD_DEVICE_ID/c8y-configuration-plugin.toml`,
+      and starts to used it for an updated list of configuration files
+      (to which this file `$TEDGE_CONFIG_DIR/c8y/$CHILD_DEVICE_ID/c8y-configuration-plugin.toml`
+      is implicitly added with type `c8y-configuration-plugin`).
+   1. If the request were triggered by the cloud,
+      the `c8y_configuration_plugin` proceeds on processing the `successful` status message,
+      notifying the cloud and removing the file `$TEDGE_CONFIG_DIR/c8y/$CHILD_DEVICE_ID/c8y-configuration-plugin.toml`.
+      Note that the request might have been triggered by the main device on start,
+      or by the child device itself (see below),
+      in that case the cloud most not be notified. 
+1. On start, the `c8y_configuration_plugin` check for each child device `$CHILD_DEVICE_ID`,
+   listed in the directory `$TEDGE_CONFIG_DIR/c8y/`,
+   if there is a known `$CHILD_DEVICE_ID/c8y-configuration-plugin.toml`.
+   1. If the configuration is missing for some `$CHILD_DEVICE_ID`
+      (i.e. if the directory `$TEDGE_CONFIG_DIR/c8y/$CHILD_DEVICE_ID` exists
```

Comment:
Still, I am not sure who creates `$TEDGE_CONFIG_DIR/c8y/$CHILD_DEVICE_ID`. If `--init <child_id>` creates it, this case (the directory exists, but the file doesn't exist) can really occur.

Reply:
> Still, I am not sure who creates `$TEDGE_CONFIG_DIR/c8y/$CHILD_DEVICE_ID`

I propose here to use `c8y_configuration_plugin --init` for that purpose.

> If `--init <child_id>` creates it, this case (the directory exists, but the file doesn't exist) can really occur.

Yes, this case can occur and the `c8y_configuration_plugin` must be prepared to handle it.

* The directory means that the child device supports configuration management.
* The file `$CHILD_DEVICE_ID/c8y-configuration-plugin.toml` tells the details on how.

Reply:
Reading this comment from @rina23q + that [other one](https://github.com/thin-edge/thin-edge.io/pull/1417#discussion_r973956217) from @albinsuresh, I wonder that we might come with a simpler solution
that is furthermore closer to what is proposed by @cstoidner for child device provisioning.

* Get rid of my proposal for an extension of the `--init` option that can do only half of the work (i.e declaring support for configuration but not providing the list of configuration files on the child device).
* Get also rid of my proposal to have the `c8y_configuration_plugin` requesting the child device for its list of configuration files if missing on the main device.
* Give the child device agent the responsibility to send its `c8y_configuration_plugin.toml` file on start and on change.
  Receiving this content, the `c8y_configuration_plugin` on the main device will have to:
  1. Declare to the mapper that the child device support configuration management (unless already done).
  2. Copy the child's `c8y_configuration_plugin.toml` under `$TEDGE_CONFIG_DIR/c8y/$CHILD_DEVICE_ID`.

@albinsuresh and @rina23q, what do you think about that?

Reply:
The child device sending all that information to the thin-edge device on startup seems like the right long term solution. But, if we want to go with a simpler short term solution, we could still use this `--init <child-id>` idea. I'd even say that let the init command create an empty `c8y-config-plugin.toml` under `$TEDGE_CONFIG_DIR/c8y/$CHILD_DEVICE_ID`. That way, the config list can even be updated from the cloud, without expecting the child device to upload that.

The main issue with that approach is the need for running this init command on the thin-edge device when a new child device is connected. Many of our customers have explicity stated that they would like a solution where child device can just be plugged-in without having to do anything on the thin-edge device. That's why I was saying that the child device initiating all this might be the better long term solution.

Reply:
I'm happy with the new proposal, `c8y_cofiguration_plugin` creates the directory `$TEDGE_CONFIG_DIR/c8y/$CHILD_DEVICE_ID` on the reception of `c8y-configuration-plugin.toml`.